### PR TITLE
Do not call uncaught exception handlers at exiting state.

### PR DESCRIPTION
**Pull request by LaszloLango**
IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
It did cause some confusion and misunderstanding that the 'process.exit' prints `uncaughtException: Abort script`. This patch fixes this. No need to call `iotjs_uncaught_exception`, when we call `process.exit()`.

**Comment by zherczeg** [comment]
I think this is still a wrong fix for this issue :)

The proper fix is implementing a watchdog. The vm timeout callback should throw an error when the timeout limit is reached, not when the engine is in exit status. The two things has nothing to do with each other. We also need to add several watchdog resets to the code. The timeout default (in seconds) should be defined in compile time, but runtime set should be available as well.

**Comment by LaszloLango** [comment]
@zherczeg I do not agree. This patch has nothing to do with watchdog. They are two different things.

File: src/iotjs.c [hunk 1]
```diff
@@ -109,7 +109,8 @@ static bool iotjs_run(iotjs_environment_t* env) {
   }
 #endif
 
-  if (throws) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_environment_t, env);
+  if (throws && (_this->state) != kExiting) {
```
Comment (daeyeon) [inline src/iotjs.c]: `iotjs_environment_is_exiting` can simplify this.

**Comment by zherczeg** [comment]
What you do: ignore exception when the engine exits. But this is not correct since even during exit an error can be thrown.

The problem is: when the engine exits, the current watchdog implementation aborts all js execution, including cleanup code. Hence if we do any cleanup in JS, an error is thrown by watchdog, and the cleanup does not run (bad for network cleanup), and we return early.

**Comment by glistening** [comment]
@zherczeg @LaszloLango #1245 might be also related to this issue. Nodejs can exit early because it works on Linux or Windows, and so on. But in our cases (NuttX and TizenRT), we have to take care of clean up by ourselves. I will make a issue for problematic cases.

**Pull request merged**